### DISCUSSION TITLE
Fast converge to oversized models

### DISCUSSION THREAD

**Discussion by HighExecutor** [keras-team/autokeras#1571] [Q&A]
Good day!

My train data has following shape: _7373 images 120x120_.
I'm trying to restrict model size with `max_model_size` parameter. For example:

```
input_node = ak.ImageInput()
output_node = ak.ConvBlock()(input_node)
output_node = ak.DenseBlock()(output_node)
output_node = ak.ClassificationHead(num_classes=10,
                                    multi_label=False,
                                    loss=None,
                                    metrics=None,)(output_node)

clf = ak.AutoModel(
    inputs=input_node,
    outputs=output_node,
    max_trials=100,
    directory=None,
    objective="val_loss",
    overwrite=True,
    max_model_size=5000000
)
```
However, the search process is converging very fast (2-5 trials) to models that have more parameters with log:

```
Oversized model: 11977642 parameters -- skipping
Oversized model: 11977642 parameters -- skipping
....
RuntimeError: Too many consecutive oversized models.
```

I played with bigger amount of parameters like 20KK and everytime I get this error. This happens for both _'greedy'_ and _'bayessian'_ tuners.
When I remove this restriction, the framework just starts to fit very big models that crucially slow the learning process or even just went out of memory. The tests were conducted on local 32GB RAM with NVidia 2080RTX machine and on DGX cluster.

**The question is, how to restrict a size of model?**
**Is this limitation for max_model_size is used by tuner to restrict the search space?**

**The Idea:**
It would be great if tuner will able to use this limitation and do not find oversized models.

**Comment by haifeng-jin** [comment]
This feature is not working that well for AutoKeras.
It may only work with large values.

The `max_model_size` would not affect how the tuner generate hyperparameter values, but only fails those models with larger sizes.
